src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -327,8 +318,20 @@ public void postBuildStatus(@NonNull BitbucketBuildStatus status) throws IOExcep
      * {@inheritDoc}
      */
     @Override
-    public boolean checkPathExists(@NonNull String branch, @NonNull String path) throws IOException {
-        return HttpStatus.SC_OK == getRequestStatus(String.format(API_BROWSE_PATH, getUserCentricOwner(), repositoryName, path, branch));
+    public boolean checkPathExists(@NonNull String branchOrHash, @NonNull String path) throws IOException {
+        StringBuilder encodedPath = new StringBuilder(path.length() + 10);
+        boolean first = true;
+        for (String segment : StringUtils.split(path, "/")) {
```

Comment:
Why is this needed? Is the `path` parameter including encoded slashes? When did that start to happen?